src/ax_interface/mib.py

```diff
@@ -37,7 +41,18 @@ def __init__(self):
             else:
                 self.update_counter += 1
             # run the background update task
-            self.update_data()
+            try:
+                self.update_data()
+            except (KeyboardInterrupt, SystemExit):
+                # They are exception thrown when Ctrl+C or sys.exit() happens
+                raise
+            except:
+                # Any other exception or error, log it and keep running
+                ex = sys.exc_info()[0]
+                exstr = "MIBUpdater.start() caught an unexpected exception: {}".format(str(ex))
+                print(exstr, flush=True)
+                logger.exception(exstr)
```

Comment:
`logger.exception` already captures the exc_info. What is here will print the exception, and have a logger message that will include the exception twice.
```python
logger.exception(exstr) 
```

Combining `print` with `logging` is a code smell.
If you want errors to go to stdout, ``addHandler(logging.StreamHandler(sys.stdout))` to https://github.com/Azure/sonic-snmpagent/blob/master/src/sonic_ax_impl/__main__.py#L39